src/run.ts

```diff
@@ -0,0 +1,272 @@
+  let octokit = new github.GitHub(githubToken);
+  let [publishCommand, ...publishArgs] = script.split(/\s+/);
+
+  let changesetPublishOutput = await execWithOutput(
+    publishCommand,
+    publishArgs,
+    { cwd }
+  );
+
+  await gitUtils.pullBranch(branch);
+  await gitUtils.push(branch, { includeTags: true });
+
+  let { packages, tool } = await getPackages(cwd);
+  let releasedPackages: Package[] = [];
+
+  if (tool !== "root") {
+    let newTagRegex = /New tag:\s+(@[^/]+\/[^@]+|[^/]+)@([^\s]+)/;
+    let packagesByName = new Map(packages.map((x) => [x.packageJson.name, x]));
+
+    for (let line of changesetPublishOutput.stdout.split("\n")) {
+      let match = line.match(newTagRegex);
+      if (match === null) {
+        continue;
+      }
+      let pkgName = match[1];
+      let pkg = packagesByName.get(pkgName);
+      if (pkg === undefined) {
+        throw new Error(
+          `Package "${pkgName}" not found.` +
+            "This is probably a bug in the action, please open an issue"
+        );
+      }
+      releasedPackages.push(pkg);
+    }
+
+    await Promise.all(
+      releasedPackages.map((pkg) =>
+        createRelease(octokit, {
+          pkg,
+          tagName: `${pkg.packageJson.name}@${pkg.packageJson.version}`,
+        })
+      )
+    );
+  } else {
+    if (packages.length === 0) {
+      throw new Error(
+        `No package found.` +
+          "This is probably a bug in the action, please open an issue"
+      );
+    }
+    const pkg = packages[0];
+    releasedPackages.push(pkg);
+    await createRelease(octokit, {
+      pkg,
+      tagName: `v${pkg.packageJson.version}`,
+    });
+  }
+
+  if (releasedPackages.length) {
+    return {
+      published: true,
+      publishedPackages: releasedPackages.map((pkg) => ({
+        name: pkg.packageJson.name,
+        version: pkg.packageJson.version,
+      })),
+    };
+  }
+
+  return { published: false };
+}
+
+type VersionOptions = {
+  script?: string;
+  githubToken: string;
+  cwd?: string;
+  prTitle?: string;
+  commitMessage?: string;
+  hasPublishScript?: boolean;
+};
+
+export async function runVersion({
+  script,
+  githubToken,
+  cwd = process.cwd(),
+  prTitle = "Version Packages",
+  commitMessage = "Version Packages",
+  hasPublishScript = false,
+}: VersionOptions) {
+  let repo = `${github.context.repo.owner}/${github.context.repo.repo}`;
+  let branch = github.context.ref.replace("refs/heads/", "");
+  let versionBranch = `changeset-release/${branch}`;
+  let octokit = new github.GitHub(githubToken);
+  let { preState } = await readChangesetState(cwd);
+
+  await gitUtils.switchToMaybeExistingBranch(versionBranch);
+  await gitUtils.reset(github.context.sha);
+
+  let versionsByDirectory = await getVersionsByDirectory(cwd);
+
+  if (script) {
+    let [versionCommand, ...versionArgs] = script.split(/\s+/);
+    await exec(versionCommand, versionArgs, { cwd });
+  } else {
+    // technically
```

Comment:
?

Reply:
not sure what I have wanted to write here 😅 gonna remove this in a moment